R/model-modules.R

```diff
@@ -247,6 +241,28 @@ enw_report <- function(non_parametric = ~0, structural = ~0, data) {
     prefix = "rep", drop_intercept = TRUE
   )
 
+  # Check for structural model and define
+  if (as_string_formula(structural) != "~0") {
+    cli::cli_alert_warning(
+      "The structural reporting model is in experimental development"
+    )
+    if (!(length(structural) == data$groups)
+      || !(length(structural[[1]]) == data$time)
+      || !(all(dim(structural[[1]][[1]]) == rep(data$max_delay, 2)))) {
```

Comment:
Implemented a basic check here for the format of `structural` but I would have to assume there's a better way to do it